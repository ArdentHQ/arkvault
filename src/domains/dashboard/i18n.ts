export const translations = {
	DASHBOARD_PAGE: {
		TITLE: "Dashboard",
	},
	FILTER_WALLETS: {
		CRYPTOASSET: {
			DESCRIPTION: "Select the networks to show.",
			TITLE: "Filter Networks",
		},
		WALLETS: {
			DESCRIPTION: "Select which wallets to view.",
			TITLE: "Your Wallets",
		},
	},

	LATEST_TRANSACTIONS: {
		EMPTY_MESSAGE:
			"You don't have any transactions yet. Once transactions have been made they will show up in your history.",
		NO_RESULTS: "No <bold>{{type}}</bold> transactions could be found.",
		TITLE: "Latest Transactions",
	},

	PORTFOLIO_BREAKDOWN: {
		FILTERED: "Please enable at least one public network to display your portfolio report.",
	},

	PORTFOLIO_BREAKDOWN_DETAILS: {
		TITLE: "Portfolio Breakdown",
	},

	WALLET_CONTROLS: {
		CREATE: "Create",
		EMPTY_MESSAGE: "Your portfolio is currently empty. <1>Create</1> or <3>Import</3> a wallet to get started.",
		EMPTY_MESSAGE_FILTERED: "You don't have any wallets matching the selected networks.",
		EMPTY_MESSAGE_TYPE: "You don't have any <bold>{{type}}</bold> wallets.",
		EMPTY_MESSAGE_TYPE_FILTERED: "You don't have any <bold>{{type}}</bold> wallets matching the selected networks.",
		IMPORT: "Import",
		IMPORT_LEDGER: "Import Ledger",
	},
};
