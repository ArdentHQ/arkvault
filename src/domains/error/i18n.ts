export const translations = {
	APPLICATION: {
		DESCRIPTION: "It looks like something went wrong. Try visiting another page or reloading.",
		HELP_TEXT: "Please provide the error below when contacting support.",
		RELOAD: "Reload",
		TITLE: "Something went wrong",
	},
	OFFLINE: {
		DESCRIPTION: "It looks like there may be a problem connecting to the Internet. Check your connection.",
		TITLE: "Connectivity Issue",
	},
};
