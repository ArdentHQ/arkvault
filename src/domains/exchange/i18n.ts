export const translations = {
	EMPTY_MESSAGE: "You haven't made any swaps yet. Once you initiate a transaction they will show up here.",

	ERROR: {
		GENERIC: "Your order could not be placed",
		INVALID_ADDRESS: "The provided address is not a valid {{ticker}} address",
		INVALID_REFUND_ADDRESS: "The provided refund address is not a valid {{ticker}} address",
	},

	EXCHANGE_FORM: {
		AMOUNT_RECEIVED: "Amount Received",
		AMOUNT_SENT: "Amount Sent",
		CURRENCY_INPUT: "{{currency}} Input",
		CURRENCY_OUTPUT: "{{currency}} Output",
		ESTIMATED_ARRIVAL: "Estimated Arrival",
		ESTIMATED_RATE: "Estimated Rate",
		ESTIMATED_TIME: "â‰ˆ {{estimatedTime}} minutes",
		EXCHANGE_ADDRESS: "Exchange {{currency}} Address",
		FROM_CURRENCY: "From Currency",
		INPUT_TRANSACTION_ID: "Input Transaction ID",
		OUTPUT_TRANSACTION_ID: "Output Transaction ID",
		PAYIN_AMOUNT: "Payin Amount",
		PAYOUT_AMOUNT: "Payout Amount",
		RECIPIENT_PLACEHOLDER: "Enter the payout address",
		RECIPIENT_WALLET: "Recipient Wallet",
		REFUND_PLACEHOLDER: "Enter the refund address",
		REFUND_WALLET: "Refund Wallet",
		SUPPORT_INFO:
			"If you have any question about your exchange, please contact {{exchange}} support team via email at <linkEmail>{{email}}</linkEmail>",
		TERMS: "I've read and agree to the {{exchange}} <linkTerms>{{terms}}</linkTerms> and <linkPrivacyPolicy>{{privacy}}</linkPrivacyPolicy>",
		TO_CURRENCY: "To Currency",
		TRANSACTION_COMPLETE: "Transaction is Complete!",
		YOUR_ADDRESS: "Your {{currency}} Address",
		YOU_GET: "You Get",
		YOU_SEND: "You Send",
	},

	EXCHANGE_NAME: "Exchange Name",
	EXPECTED_AMOUNT_HINT: "The final payout amount might be different",

	MODAL_DELETE_EXCHANGE_TRANSACTION: {
		DESCRIPTION:
			"Are you sure you want to remove this swap from the transaction history? This action cannot be undone.",
		TITLE: "Remove Transaction",
	},

	NAVIGATION: {
		EXCHANGES: "Exchanges",
		TRANSACTIONS: "Transactions",
	},

	PAGE_EXCHANGES: {
		ADD_EXCHANGE: "Add Exchange",
		DELETE_CONFIRMATION: "The exchange transaction <bold>{{orderId}}</bold> has been deleted",
		EMPTY_MESSAGE: "There are currently no exchange providers available, please try again at a later time.",
		SUBTITLE: "Choose one of the exchange providers below to swap directly from within the wallet.",
		TITLE: "Exchange",
	},

	REFUND_WALLET: {
		ADD: "Add Refund Wallet",
		REMOVE: "Remove Refund Wallet",
	},

	SELECT_EXCHANGE_MESSAGE: "Select one of the exchanges you have installed",

	STATUS: {
		CONFIRMING: "Confirming",
		EXCHANGING: "Exchanging",
		EXPIRED: "Expired",
		FAILED: "Failed",
		FINISHED: "Finished",
		NEW: "New",
		REFUNDED: "Refunded",
		SENDING: "Sending",
		VERIFYING: "Verifying",
		WAITING: "Waiting",
	},

	TO_ADDRESS: "To Address",

	TRANSACTION_ID: "Transaction ID",
	TRANSACTION_STATUS: {
		AWAITING_DEPOSIT: "Awaiting Deposit",
		EXCHANGING: "Exchanging",
		EXPIRED: "This transaction has expired.",
		FAILED: "This transaction has failed.",
		REFUNDED: "This transaction has been refunded.",
		SENDING: "Sending",
		VERIFYING: "This transaction is being verified by the provider.",
	},

	VALIDATION: {
		INVALID_ADDRESS: "Not a valid {{ticker}} address",
		MIN_AMOUNT: "The minimal exchange amount is {{amount}} {{ticker}}",
		PAIR_NOT_AVAILABLE: "The pair <bold>{{from}}</bold> / <bold>{{to}}</bold> is not available.",
	},

	YOUR_EXCHANGE_LIST: "Your Exchange List",
};
