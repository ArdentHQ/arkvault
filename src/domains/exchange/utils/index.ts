export * from "./assertions";
export * from "./errors";
