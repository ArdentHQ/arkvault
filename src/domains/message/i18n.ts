export const translations = {
  LEDGER_CONFIRMATION: {
    TITLE: "Confirm Your Signature",
  },

	PAGE_SIGN_MESSAGE: {
      AUTHENTICATION_STEP: {
      DESCRIPTION_ENCRYPTION_PASSWORD: "Enter your encryption password to sign the message.",
      DESCRIPTION_MNEMONIC: "Enter your mnemonic passphrase to sign the message.",
      DESCRIPTION_SECRET: "Enter your secret to sign the message.",
    },

		COPY_SIGNATURE: "Copy Signature",

		FORM_STEP: {
			DESCRIPTION_ENCRYPTION_PASSWORD: "Provide a message below and sign with your encryption password.",
			DESCRIPTION_LEDGER: "Provide a message below and sign with your ledger.",
			DESCRIPTION_MNEMONIC: "Provide a message below and sign with your mnemonic passphrase.",
			DESCRIPTION_SECRET: "Provide a message below and sign with your secret.",
			TITLE: "Sign Message",
		},

		SIGN: "Sign",

		SUCCESS_STEP: {
      TITLE: "Message Successfully Signed",
		},

		TITLE: "Sign Message",
	},

	PAGE_VERIFY_MESSAGE: {
		TITLE: "Verify Message",
	},

	SIGNATORY: "Signatory",
};
