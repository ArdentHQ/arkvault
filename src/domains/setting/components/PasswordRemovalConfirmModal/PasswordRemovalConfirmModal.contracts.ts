interface PasswordRemovalProperties {
	onCancel: () => void;
	onConfirm: (currentPassword: string) => void;
}

interface PasswordRemovalFormState {
	currentPassword: string;
}

export type { PasswordRemovalProperties, PasswordRemovalFormState };
