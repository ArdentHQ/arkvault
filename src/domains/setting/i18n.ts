export const translations = {
	APPEARANCE: {
		MENU_ITEM: "Appearance",
		OPTIONS: {
			ACCENT_COLOR: {
				COLORS: {
					BLUE: "Navy",
					GREEN: "Green",
				},
				DESCRIPTION: "Choose a color for buttons and other wallet features.",
				TITLE: "Accent Color",
			},
			EXPANDED_TABLES: {
				DESCRIPTION: "Increases spacing of table items.",
				TITLE: "Expanded Tables",
			},
			LATEST_TRANSACTIONS: {
				DESCRIPTION: "Enables latest transactions on Portfolio page.",
				TITLE: "Portfolio Latest Transactions",
			},
			VIEWING_MODE: {
				DESCRIPTION: "Choose a light or dark viewing mode.",
				TITLE: "Viewing Mode",
				VIEWING_MODES: {
					DARK: "Dark",
					LIGHT: "Light",
				},
			},
			WALLET_NAMING: {
				DESCRIPTION: "Prioritize showing network provided names over wallet names.",
				TITLE: "Wallet Naming",
			},
		},
		SUBTITLE: "Customize the look of the application.",
		TITLE: "Appearance",
	},
	EXPORT: {
		ADD_INFORMATION_ABOUT_THE_NETWORK: "Add information about the network",
		MENU_ITEM: "Export",
		OPTIONS: {
			EXCLUDE_EMPTY_WALLETS: {
				DESCRIPTION: "Wallets with a balance of 0 will not be exported.",
				TITLE: "Exclude Empty Wallets",
			},
			EXCLUDE_LEDGER_WALLETS: {
				DESCRIPTION: "Ledger hardware addresses will not be exported.",
				TITLE: "Exclude Ledger Wallets",
			},
		},
		SAVE_GENERAL_CUSTOMIZATIONS: "Save general settings customizations",
		SUBTITLE:
			"Your exported Profile will not contain your mnemonic passphrases, only addresses and respective names.",
		SUCCESS: "Your settings have been exported successfully",
		TITLE: "Export Profile",
	},

	GENERAL: {
		ERROR: "Your profile settings could not be updated",
		MENU_ITEM: "General",
		OTHER: {
			DARK_THEME: {
				DESCRIPTION: "Enables a dark, high contrast scheme.",
				TITLE: "Dark Theme",
			},
			DEVELOPMENT_NETWORKS: {
				DESCRIPTION:
					"To create or import wallets from development and test networks, you must first enable this feature.",
				TITLE: "Development and Test Networks",
			},
			RESET_SETTINGS: {
				DESCRIPTION: "Restore Profile settings to the default.",
				TITLE: "Reset Settings",
			},
			TITLE: "Other",
		},
		PERSONAL: {
			CONFIRM_PASSWORD: "Confirm Password",
			CURRENCY: "Currency",
			LANGUAGE: "Language",
			MARKET_PROVIDER: "Market Data Provider",
			NAME: "Profile Name",
			PASSPHRASE_LANGUAGE: "Passphrase Language",
			PASSWORD: "Password",
			PROFILE_IMAGE: "Profile Image",
			REMOVE_AVATAR: "Remove Avatar",
			TIME_FORMAT: "Time Format",
			TITLE: "Personal Details",
			UPLOAD_AVATAR: "Upload Avatar",
			VALIDATION: {
				NAME_EXISTS: "Profile name already exists",
			},
		},
		SECURITY: {
			ADVANCED_MODE: {
				DESCRIPTION: "Allows installations from direct URLs. Enable at your own risk.",
				TITLE: "Advanced Mode",
			},
			AUTOMATIC_SIGN_OUT_PERIOD: {
				TITLE: "Auto Log-Off",
			},
			TITLE: "Security",
		},
		SUBTITLE: "Customize your wallet to suit your needs.",
		SUCCESS: "Your profile settings have been updated",
		TITLE: "Profile Settings",
		UNSUPPORTED_CURRENCY: "{{currency}} is not supported by {{provider}}. Currency has been changed to USD.",
	},

	MODAL_DEVELOPMENT_NETWORK: {
		DESCRIPTION:
			"Disabling this setting will hide your wallets associated with development networks. Are you sure you want to continue?",
		TITLE: "Development Network",
	},

	NETWORKS: {
		ADD_NEW_NETWORK: {
			DESCRIPTION: "Add a custom network by providing a valid seed address.",
			TITLE: "Add Network",
		},
		CUSTOM_NETWORKS: {
			ADD_NETWORK: "Add Network",
		},
		DELETE_MODAL: {
			DESCRIPTION:
				"This action cannot be undone and will result in  removal of all wallets and contact addresses that  were added for this network. To confirm this action, enter the network name below.",
			TITLE: "Remove Network",
		},
		DETAILS_MODAL: {
			DESCRIPTION: "Overview of the data obtained from the network.",
			TITLE: "Network Information",
		},
		FORM: {
			DELETE_CONFIRM_PLACEHOLDER: "Enter the network name to confirm removal",
			EXPLORER: "Explorer",
			FETCHING_ERROR: "Cannot obtain network data. Check Seed Server.",
			INVALID_KNOWN_WALLETS_URL: "Invalid Known Wallets URL",
			INVALID_MARKET_TICKER_FORMAT: "Incorrect market ticker format",
			INVALID_SEED_SERVER_FORMAT: "Invalid seed server format",
			INVALID_SLIP_FORMAT: "Invalid SLIP (BIP44) format",
			KNOWN_WALLETS: "Known Wallets URL",
			MARKET_TICKER: "Market Ticker",
			NETWORK_HASH_MISMATCH:
				"Seed server you are trying to add is not part of this network. Add new custom network or provide correct seed server for current network.",
			NETWORK_NAME: "Network Name",
			SEED_SERVER: "Seed Server",
			SEED_SERVER_PLACEHOLDER: "https://ip:port or https://domain",
			SLIP: "SLIP (BIP44)",
		},
		MENU_ITEM: "Networks",
		MESSAGES: {
			AT_LEAST_ONE_PUBLIC_NETWORK: "At least 1 public network required.",
			AT_LEAST_ONE_TEST_NETWORK: "At least 1 development or test network required",
			CUSTOM_NETWORK_ADDED: "Custom network <strong>{{- networkName}}</strong> was successfully added.",
			CUSTOM_NETWORK_DELETED:
				"<strong>{{- networkName}}</strong> has been successfully deleted from your profile.",
		},
		OPTIONS: {
			CUSTOM_NETWORKS: {
				DESCRIPTION: "Add and manage Custom Networks for your Profile.",
				TITLE: "Custom Networks",
			},
			DEVELOPMENT_AND_TEST_NETWORKS: {
				DESCRIPTION: "Manage the Development and Test Networks for your Profile.",
				TITLE: "Development and Test Networks",
			},
			PUBLIC_NETWORKS: {
				DESCRIPTION: "Manage Public Networks for your Profile.",
				TITLE: "Public Networks",
			},
		},
		SUBTITLE: "Customize and manage networks. Enabled and selected networks will be visible in your Profile.",
		TITLE: "Manage Networks",
		UPDATE_NETWORK: {
			DESCRIPTION: "Modify settings of the selected network.",
			TITLE: "Edit Network",
		},
	},

	PASSWORD: {
		BUTTON: {
			CHANGE: "Change Password",
			CREATE: "Set Password",
			REMOVE: "Remove Password",
		},
		CURRENT: "Current Password",
		ERROR: {
			FALLBACK: "Something went wrong.",
			MISMATCH: "The current password does not match.",
		},
		MENU_ITEM: "Password",
		PASSWORD_1: "New Password",
		PASSWORD_2: "Confirm New Password",
		REMOVAL: {
			DESCRIPTION:
				"Removing your profile password will result in anyone with access to your Payvo Wallet to see all your profile data. Are you sure you wish to remove it?",
			PROFILE_PASSWORD: "Profile Password",
			SUCCESS: "Password successfully removed.",
			TITLE: "Remove Profile Password",
		},
		SUBTITLE: {
			CHANGE: "Change the password used to secure your Profile below.",
			CREATE: "Set your password below to secure your Profile.",
		},
		SUCCESS: "Your password has been successfully changed.",
		TITLE: "Profile Password",
	},

	SERVERS: {
		ADD_NEW_SERVER: {
			DESCRIPTION:
				"Add a custom network peer or multisignature server for the selected network by providing a valid address.",
			FETCHING_DETAILS: "Fetching details ...",
			FETCHING_ERROR:
				"We were unable to connect to the provided server or failed to find the expected information. Please check your input and try again.",
			MULTISIG_SERVER: "Multisig Server",
			NETWORK_MISMATCH_ERROR: "Failed to connect to host because it is on another network",
			NETWORK_PLACEHOLDER: "https://ip:port/api or https://domain",
			PEER_SERVER: "Network Peer",
			SUCCESS_MESSAGE:
				"New {{networkType}} <strong>{{serverName}}</strong> for <strong>{{networkName}}</strong> has been successfully added.",
			TITLE: "Add New Server",
		},
		CUSTOM_PEERS: {
			EMPTY_MESSAGE: "No custom network peers or multisignature servers added yet.",
		},
		DELETE_MODAL: {
			DESCRIPTION:
				"By removing <strong>{{serverName}}</strong> your interactions with <strong>{{networkName}}</strong> might become unavailable if other servers are unreachable. Are you sure you wish to remove it?",
			TITLE: "Remove Server",
		},
		EDIT_SERVER: {
			DESCRIPTION: "Edit information for custom added server.",
			SUCCESS_MESSAGE: "Server data for <strong>{{serverName}}</strong> has been successfully updated.",
			TITLE: "Edit Server",
		},
		MENU_ITEM: "Servers",
		NODE_STATUS_TOOLTIPS: {
			HEALTHY: "Default nodes are healthy.",
			WITH_ISSUES: "Default nodes are experiencing issues, please check on socials for more information.",
		},
		OPTIONS: {
			CUSTOM_PEERS: {
				DESCRIPTION:
					"Manage custom peers and multisig servers. Using more than one per network will broadcast to all selected.",
				TITLE: "Custom Peers and Multisig Servers",
			},
			DEFAULT_NODE_STATUS: {
				DESCRIPTION: "Check health of default nodes for each enabled network.",
				TITLE: "Default Nodes Status",
			},
			FALLBACK_TO_DEFAULT_NODES: {
				DESCRIPTION:
					"Only used when custom servers are selected. In case of issues, the wallet will fallback to using the default nodes to broadcast transactions.",
				TITLE: "Fallback to Default Nodes",
			},
		},
		PEERS_STATUS_TOOLTIPS: {
			HEALTHY: "Peer is healthy",
			WITH_ISSUES: "Peer is not resolving",
		},
		SUBTITLE: "Manage custom network peers and multisignature servers for your Profile.",
		SUCCESS: "Your server settings has been successfully updated.",
		TITLE: "Peers & Multisig Servers",
	},

	TITLE: "Settings",
};
