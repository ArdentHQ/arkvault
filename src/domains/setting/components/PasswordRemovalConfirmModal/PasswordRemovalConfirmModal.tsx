import React from "react";
import { SubmitHandler, useForm } from "react-hook-form";
import { useTranslation } from "react-i18next";

import { Alert } from "@/app/components/Alert";
import { Button } from "@/app/components/Button";
import { Form, FormButtons, FormField, FormLabel } from "@/app/components/Form";
import { Icon } from "@/app/components/Icon";
import { Image } from "@/app/components/Image";
import { InputPassword } from "@/app/components/Input";
import { Modal } from "@/app/components/Modal";

import { PasswordRemovalFormState, PasswordRemovalProperties } from "./PasswordRemovalConfirmModal.contracts";

export const PasswordRemovalConfirmModal: React.FC<PasswordRemovalProperties> = ({ onCancel, onConfirm }) => {
	const { t } = useTranslation();

	const form = useForm<PasswordRemovalFormState>({
		defaultValues: {
			currentPassword: "",
		},
		mode: "onChange",
	});

	const { formState, register } = form;
	const { isDirty, isValid } = formState;

	const submit: SubmitHandler<PasswordRemovalFormState> = ({ currentPassword }) => {
		onConfirm(currentPassword);
	};

	return (
		<Modal
			isOpen
			title={t("SETTINGS.PASSWORD.REMOVAL.TITLE")}
			image={<Image name="Trash" useAccentColor={false} className="mx-auto my-8 max-w-52" />}
			onClose={onCancel}
		>
			<Alert>{t("SETTINGS.PASSWORD.REMOVAL.DESCRIPTION")}</Alert>

			<Form onSubmit={submit} context={form} className="mt-8 space-y-8">
				<FormField name="currentPassword">
					<FormLabel label={t("SETTINGS.PASSWORD.REMOVAL.PROFILE_PASSWORD")} />
					<InputPassword
						data-testid="PasswordRemovalConfirmModal__input-password"
						ref={register({
							required: t("COMMON.VALIDATION.FIELD_REQUIRED", {
								field: t("SETTINGS.PASSWORD.CURRENT"),
							}).toString(),
						})}
					/>
				</FormField>

				<FormButtons>
					<Button data-testid="PasswordRemovalConfirmModal__cancel" onClick={onCancel} variant="secondary">
						{t("COMMON.CANCEL")}
					</Button>

					<Button
						data-testid="PasswordRemovalConfirmModal__confirm"
						variant="danger"
						className="flex space-x-2"
						disabled={isDirty ? !isValid : true}
						type="submit"
					>
						<Icon name="Trash" />
						<span>{t("COMMON.REMOVE")}</span>
					</Button>
				</FormButtons>
			</Form>
		</Modal>
	);
};
