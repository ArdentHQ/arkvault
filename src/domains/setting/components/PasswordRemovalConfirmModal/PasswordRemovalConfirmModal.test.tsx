import userEvent from "@testing-library/user-event";
import React from "react";

import { PasswordRemovalConfirmModal } from "./PasswordRemovalConfirmModal";
import { buildTranslations } from "@/app/i18n/helpers";
import { render, screen, waitFor } from "@/utils/testing-library";

const translations = buildTranslations();

describe("PasswordRemovalConfirmModal", () => {
	it("should render", async () => {
		const { asFragment } = render(<PasswordRemovalConfirmModal onCancel={jest.fn()} onConfirm={jest.fn()} />);

		await expect(screen.findByText(translations.SETTINGS.PASSWORD.REMOVAL.PROFILE_PASSWORD)).resolves.toBeVisible();

		expect(asFragment()).toMatchSnapshot();
	});

	it("should call onCancel when closed", async () => {
		const onCancel = jest.fn();

		render(<PasswordRemovalConfirmModal onCancel={onCancel} onConfirm={jest.fn()} />);

		await expect(screen.findByText(translations.SETTINGS.PASSWORD.REMOVAL.PROFILE_PASSWORD)).resolves.toBeVisible();

		userEvent.click(screen.getByTestId("PasswordRemovalConfirmModal__cancel"));

		expect(onCancel).toHaveBeenCalledWith(expect.objectContaining({ nativeEvent: expect.any(MouseEvent) }));
	});

	it("should call onConfirm when submitted", async () => {
		const onConfirm = jest.fn();

		render(<PasswordRemovalConfirmModal onCancel={jest.fn()} onConfirm={onConfirm} />);

		await expect(screen.findByText(translations.SETTINGS.PASSWORD.REMOVAL.PROFILE_PASSWORD)).resolves.toBeVisible();

		expect(screen.getByTestId("PasswordRemovalConfirmModal__confirm")).toBeDisabled();

		userEvent.paste(screen.getByTestId("PasswordRemovalConfirmModal__input-password"), "password");

		await waitFor(() => expect(screen.getByTestId("PasswordRemovalConfirmModal__confirm")).not.toBeDisabled());

		userEvent.click(screen.getByTestId("PasswordRemovalConfirmModal__confirm"));

		await waitFor(() => expect(onConfirm).toHaveBeenCalledWith("password"));
	});
});
