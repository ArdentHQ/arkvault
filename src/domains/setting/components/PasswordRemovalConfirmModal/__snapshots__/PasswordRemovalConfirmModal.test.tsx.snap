// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordRemovalConfirmModal should render 1`] = `
<DocumentFragment>
  <div
    class="overflow-overlay fixed inset-0 z-50 flex w-full overflow-y-auto bg-theme-secondary-900-rgba bg-opacity-60 dark:bg-black-rgba dark:bg-opacity-80 md:py-20"
    data-testid="Modal__overlay"
  >
    <div
      class="css-amqh1w"
      tabindex="-1"
    >
      <div
        class="relative flex min-h-full flex-col overflow-hidden bg-theme-background shadow-2xl sm:mb-0 md:h-auto md:rounded-2.5xl px-10 pt-10 pb-24 sm:pb-10"
        data-testid="Modal__inner"
      >
        <div
          class="absolute top-0 right-0 z-10 mt-5 mr-5 rounded bg-theme-primary-100 transition-all duration-100 ease-linear hover:bg-theme-primary-300 dark:bg-theme-secondary-800 dark:text-theme-secondary-600 dark:hover:bg-theme-secondary-700 dark:hover:text-theme-secondary-400"
        >
          <button
            class="h-11 w-11 css-12xyeiw"
            data-testid="Modal__close-button"
            type="button"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                class="css-v0ob3f"
                height="16"
                width="16"
              >
                <svg>
                  cross.svg
                </svg>
              </div>
            </div>
          </button>
        </div>
        <div
          class="relative flex flex-1 flex-col space-y-4"
        >
          <h2
            class="mb-0 text-2xl font-bold"
          >
            Remove Profile Password
          </h2>
          <div
            class="flex flex-1 flex-col"
          >
            <svg
              class="my-8 mx-auto max-w-52"
            >
              trash-light.svg
            </svg>
            <div
              class="flex flex-col overflow-hidden rounded-xl"
            >
              <div
                class="css-pra96q"
              >
                <div
                  class="css-v0ob3f"
                  height="16"
                  width="16"
                >
                  <svg>
                    circle-exclamation-mark.svg
                  </svg>
                </div>
                <span>
                  Warning
                </span>
              </div>
              <div
                class="css-1pvpof3"
              >
                Removing your profile password will result in anyone with access to your device to see all your profile data. Are you sure you wish to remove it?
              </div>
            </div>
            <form
              class="mt-8 space-y-8"
              data-testid="Form"
            >
              <fieldset
                class="flex flex-col css-1bcidjh"
              >
                <label
                  class="FormLabel mb-2 flex text-sm font-semibold text-theme-secondary-text transition-colors duration-100"
                  data-testid="FormLabel"
                  for="currentPassword"
                >
                  Profile Password
                </label>
                <div
                  class="css-1psxlcd"
                >
                  <div
                    class="relative flex h-full flex-1"
                  >
                    <input
                      autocomplete="off"
                      class="no-ligatures w-full border-none css-1nupw50"
                      data-testid="PasswordRemovalConfirmModal__input-password"
                      name="currentPassword"
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    class="flex items-center space-x-3 divide-x divide-theme-secondary-300 dark:divide-theme-secondary-800 text-theme-primary-300 dark:text-theme-secondary-600"
                    data-testid="Input__addon-end"
                  >
                    <div
                      class=""
                    >
                      <button
                        class="ring-focus relative flex h-full w-full items-center justify-center text-2xl focus:outline-none"
                        data-ring-focus-margin="-m-1"
                        data-testid="InputPassword__toggle"
                        type="button"
                      >
                        <div
                          class="css-1i4b0eq"
                          height="20"
                          width="20"
                        >
                          <svg>
                            eye.svg
                          </svg>
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
              </fieldset>
              <div
                class="css-1tkbsa1"
              >
                <button
                  class="css-1jhy6v9"
                  data-testid="PasswordRemovalConfirmModal__cancel"
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    Cancel
                  </div>
                </button>
                <button
                  class="flex space-x-2 css-1yqe28v"
                  data-testid="PasswordRemovalConfirmModal__confirm"
                  disabled=""
                  type="submit"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="css-v0ob3f"
                      height="16"
                      width="16"
                    >
                      <svg>
                        trash.svg
                      </svg>
                    </div>
                    <span>
                      Remove
                    </span>
                  </div>
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
